src/fsharp/TypeChecker.fs

```diff
@@ -10005,18 +10005,20 @@ and TcMethodArg  cenv env  (lambdaPropagationInfo, tpenv) (lambdaPropagationInfo
                 let minFuncArity = overloadsWhichAreFuncAtThisPosition |> Array.minBy Array.length |> Array.length
                 let prefixOfLambdaArgsForEachOverload = overloadsWhichAreFuncAtThisPosition |> Array.map (Array.take minFuncArity)
           
-                if prefixOfLambdaArgsForEachOverload.Length > 0 then 
+                if not (Array.isEmpty prefixOfLambdaArgsForEachOverload) then 
```

Comment:
I prefer the original code on this line (but it's not big deal either way)

Reply:
Actually I only touched it to show we checked that case. Now I can revert it ;-)